### DISCUSSION TITLE
Write formulae that depends on vulkan-sdk

### DISCUSSION THREAD

**Discussion by CarterLi** [Homebrew/discussions#2957] [Writing Formulae/Casks] [answered]
More multi-platform libs have vulkan support, for example [libplacebo](https://github.com/haasn/libplacebo). We do have [molten-vk](https://github.com/Homebrew/homebrew-core/blob/master/Formula/molten-vk.rb) however most libraries requires full vulkan sdk

https://vulkan.lunarg.com/sdk/home#mac

Any suggestions?

**Comment by SMillerDev** [comment, marked as answer]
Is the sdk open source?
Yes -> make it into a formula and declare it as dependency

No ->
Is your formula in a non-homebrew tap?
Yes -> you can depend on whatever you want 

No -> non-formula dependencies are not allowed in homebrew, but you can always ship it in a personal tap.

**Reply by CarterLi** [reply]
> Is your formula in a non-homebrew tap?
> Yes -> you can depend on whatever you want

How can I make my tap find the sdk? The library uses `pkg-config` and vulkan sdk `.pc` files were installed in `/usr/local/lib/pkgconfig/`

**Reply by SMillerDev** [reply]
From https://people.freedesktop.org/~dbn/pkg-config-guide.html:
> The message above references the `PKG_CONFIG_PATH` environment variable. This variable is used to augment pkg-config's search path.

**Reply by CarterLi** [reply]
Well, no. Homebrew seems rewriting PKG_CONFIG_PATH. I can build the lib manually with vulkan support, but not with homebrew formula
